src/Umbraco.Core/Persistence/Repositories/Implement/ContentTypeRepositoryBase.cs

```diff
@@ -219,6 +219,7 @@ protected void PersistNewBaseContentType(IContentTypeComposition entity)
 
         protected void PersistUpdatedBaseContentType(IContentTypeComposition entity)
         {
+            CorrectPropertyTypeVariations(entity);
```

Comment:
As requested I moved this behavior back to ContentTypeRepositoryBase.
It is done before validation because the validation method essentially checks what is done in the correction method. As you mentioned @Shazwazza this makes the validation method essentially obsolete but does not really hurt either.

Note this "auto correct" behavior is not done (yet) in `PersistNewBaseContentType()` but only in the update method. This is how it was before any of our changes so I didn't add the "auto correct" code there, but we could add it there as well.

Reply:
All good (for now), maybe it could be added to the `PersistNewBaseContentType` but lets keep changes to a minimum for now.